platform/app/src/hooks/useSearchParams.js

```diff
@@ -1,17 +1,23 @@
-import useQuery from './useQuery';
+import { useLocation } from 'react-router';
 
 /**
- * It returns a Map of the query parameters in the URL, where the keys are
- * lowercase
- * @returns A function that returns a Map of the query parameters.
+ * It returns a URLSearchParams of the query parameters in the URL, where the keys are
+ * either lowercase or maintain their case based on the lowerCaseKeys parameter.
+ * @param {boolean} lowerCaseKeys true to return lower case keys; false (default) to maintain casing;
+ * @returns {URLSearchParams}
  */
-export default function useSearchParams() {
-  const query = useQuery();
-  // make query params case-insensitive
-  const searchParams = new Map();
-  for (const [key, value] of query) {
-    searchParams.set(key.toLowerCase(), value);
+export default function useSearchParams(lowerCaseKeys = false) {
```

Comment:
I prefer keyword arguments for two reasons 

1. less chance to break API in the future
2. it makes it usuage much more clear when reading it in another file `useSearchParams(true)` vs `useSearchParams({lowerCaseKeys: true})`